engine/src/main/java/org/hibernate/search/engine/spi/DocumentBuilderIndexedEntity.java

```diff
@@ -803,12 +809,46 @@ else if ( numericValue instanceof Float ) {
 					}
 				}
 				else {
-					document.add( new SortedDocValuesField( sortField.getFieldName(), new BytesRef( field.stringValue() ) ) );
+					String value = field.stringValue();
+					AnalyzerReference analyzerReference = fieldMetaData.getAnalyzerReference();
+					if ( fieldMetaData.getIndex().isAnalyzed() && analyzerReference.is( LuceneAnalyzerReference.class ) ) {
+						// Analysis is skipped altogether when the analyzer is remote. It's up to the backend to handle it.
+						Analyzer analyzer = analyzerReference.unwrap( LuceneAnalyzerReference.class ).getAnalyzer();
+						value = analyzeSortFieldValue( sortField.getFieldName(), field.stringValue(), analyzer );
+					}
+					if ( value != null ) {
+						document.add( new SortedDocValuesField( sortField.getFieldName(), new BytesRef( value ) ) );
+					}
+				}
+			}
+		}
+	}
+
+	/*
+	 * Necessary because Lucene doesn't automatically analyze SortedDocValuesFields.
```

Comment:
I don't think it's the right wording as we don't even provide the analyzer to the `SortedDocValuesField`s so we can't expect Lucene to deal with this.